corgidrp/astrom.py

```diff
@@ -0,0 +1,387 @@
+        raise TypeError('cal_properties must be tuple (platescale, north_angle)')
+
+    image_shape = np.shape(image)
+    image_center_x = (image_shape[0]-1) // 2
+    image_center_y = (image_shape[1]-1) // 2    
+    
+    # estimate the location of the target star with pyklip
+    # assuming the target source is meant to fall on the center pixel
+    target_guess = (image_center_x, image_center_y)
+    peakflx, fwhm, source_center_x, source_center_y = fakes.gaussfit2d(frame= image, xguess= target_guess[0], yguess= target_guess[1])
+    
+    offset_x = source_center_x - image_center_x
+    offset_y = source_center_y - image_center_y
+
+    # convert pixel offset back to SkyCoord separation
+    center_pix_RA = ((target_coordinate[0] * astropy.units.deg) - ((offset_x * cal_properties[0]) * astropy.units.mas).to(astropy.units.deg)).value
+    center_pix_DEC = ((target_coordinate[1] * astropy.units.deg) - ((offset_y * cal_properties[0]) * astropy.units.mas).to(astropy.units.deg)).value
+
+    return (center_pix_RA, center_pix_DEC)
+
+def astrometric_calibration(input_dataset, guesses, target_coordinate):
+    """
+    Perform the boresight calibration of a dataset.
+    
+    Args:
+        input_dataset (corgidrp.data.Dataset): dataset containing a single image for astrometric calibration
+        guesses (str): path to file with x,y [pixel] locations of sources AND RA,DEC [deg] true source positions
+        target_coordinate (str): path to file with RA,DEC coordinate of target source 
+        
+    Returns:
+        corgidrp.data.Dataset: a new dataset of one corgidrp.data.AstrometricCalibration() object
+        
+    """
+    if type(guesses) is not str:
+        raise TypeError('guesses must be a str')
+    else:
+        guesses = ascii.read(guesses)
+        if 'x' and 'y' and 'RA' and 'DEC' not in guesses.colnames:
+            raise ValueError('guesses must have column names [\'x\',\'y\',\'RA\',\'DEC\']')
+
+    if type(target_coordinate) is not str:
+        raise TypeError('target_coordinate must be a str')
+    else:
+        target = ascii.read(target_coordinate)
+        if 'RA' and 'DEC' not in target.colnames:
+            raise ValueError('target_coordinate must have column names [\'RA\',\'DEC\']')
+
+
+    dataset = input_dataset.copy()
+    image = dataset[0].data
+    target_coordinate = (target['RA'][0], target['DEC'][0])
+
+    platescale = compute_platescale(image=image, source_guesses=guesses, skycoords_=guesses)
+
+    northangle = compute_northangle(image=image, source_guesses=guesses, center_coord=target_coordinate, skycoords_=guesses)
+
+    cal_properties = (platescale, northangle)
+    ra, dec = compute_boresight(image=image, target_coordinate=target_coordinate, cal_properties=cal_properties)
+
+    astrom_data = np.array([ra, dec, platescale, northangle])
+    astrom_cal = corgidrp.data.AstrometricCalibration(astrom_data, pri_hdr=dataset[0].pri_hdr, ext_hdr=dataset[0].ext_hdr)
```

Comment:
you can just return the single astrometriccalibration file (with the updated history). It's ok that it isn't a dataset object, since we want to output a single calibration file at the end.